pkg/parser/parsing_test.go

```diff
@@ -98,162 +115,223 @@ func testOneParser(t *testing.T, dir string) error {
 	//TBD: Load post overflows
 	//func testFile(t *testing.T, file string, pctx UnixParserCtx, nodes []Node) bool {
 	parser_test_file := fmt.Sprintf("%s/test.yaml", dir)
-	if testFile(t, parser_test_file, *pctx, pnodes) != true {
-		return fmt.Errorf("test failed !")
+	tests := loadTestFile(parser_test_file)
+	count := 1
+	if b != nil {
+		count = b.N
+		b.ResetTimer()
+	}
+	for n := 0; n < count; n++ {
+		if testFile(tests, *pctx, pnodes) != true {
+			return fmt.Errorf("test failed !")
+		}
 	}
 	return nil
 }
 
-func testFile(t *testing.T, file string, pctx UnixParserCtx, nodes []Node) bool {
+func prepTests() (*UnixParserCtx, error) {
+	var pctx *UnixParserCtx
+	var p UnixParser
+
+	//Load enrichment
+	datadir := "../../data/"
+	pplugins, err := Loadplugin(datadir)
+	if err != nil {
+		log.Fatalf("failed to load plugin geoip : %v", err)
+	}
+	ECTX = nil
+	ECTX = append(ECTX, pplugins)
+	log.Printf("Loaded -> %+v", ECTX)
+
+	//Load the parser patterns
+	cfgdir := "../../config/"
 
-	var expects []types.Event
+	/* this should be refactored to 2 lines :p */
+	// Init the parser
+	pctx, err = p.Init(map[string]interface{}{"patterns": cfgdir + string("/patterns/")})
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize parser : %v", err)
+	}
+	return pctx, nil
+}
 
-	/* now we can load the test files */
-	//process the yaml
+func loadTestFile(file string) []TestFile {
 	yamlFile, err := os.Open(file)
 	if err != nil {
-		t.Errorf("yamlFile.Get err   #%v ", err)
+		log.Fatalf("yamlFile.Get err   #%v ", err)
 	}
 	dec := yaml.NewDecoder(yamlFile)
 	dec.SetStrict(true)
+	var testSet []TestFile
 	for {
 		tf := TestFile{}
 		err := dec.Decode(&tf)
 		if err != nil {
 			if err == io.EOF {
-				log.Warningf("end of test file")
 				break
 			}
-			t.Errorf("Failed to load testfile '%s' yaml error : %v", file, err)
-			return false
+			log.Fatalf("Failed to load testfile '%s' yaml error : %v", file, err)
+			return nil
 		}
-		for _, in := range tf.Lines {
-			log.Debugf("Parser input : %s", spew.Sdump(in))
-			out, err := Parse(pctx, in, nodes)
-			if err != nil {
-				log.Errorf("Failed to process %s : %v", spew.Sdump(in), err)
+		testSet = append(testSet, tf)
+	}
+	return testSet
+}
+
+func matchEvent(expected types.Event, out types.Event, debug bool) ([]string, bool) {
+	var retInfo []string
+	var valid bool
+	expectMaps := []map[string]string{expected.Parsed, expected.Meta, expected.Enriched}
+	outMaps := []map[string]string{out.Parsed, out.Meta, out.Enriched}
+	outLabels := []string{"Parsed", "Meta", "Enriched"}
+
+	//allow to check as well for stage and processed flags
+	if expected.Stage != "" {
+		if expected.Stage != out.Stage {
+			if debug {
+				retInfo = append(retInfo, fmt.Sprintf("mismatch stage %s != %s", expected.Stage, out.Stage))
+			}
+			valid = false
+			goto checkFinished
+		} else {
+			valid = true
```

Comment:
The tool reporting this linter error, has the pretention to not display false positive. But from what I see it is a false positive.